src/librepgp/stream-write.c

```diff
@@ -331,8 +331,16 @@ encrypted_start_aead_chunk(pgp_dest_encrypted_param_t *param, size_t idx, bool l
     STORE64BE(param->ad + param->adlen - 8, idx);
 
     if (last) {
-        total = idx ? (idx - 1) * param->chunklen : 0;
-        total += param->cachelen + param->chunkout;
+        if (param->chunkout + param->cachelen == 0) {
+            /* we need to clearly reset it since cipher was initialized but not finished */
+            pgp_cipher_aead_reset(&param->encrypt);
+        }
+
+        total = idx * param->chunklen;
+        if (param->cachelen + param->chunkout) {
+            total -= param->chunklen - param->cachelen - param->chunkout;
```

Comment:
Code is a bit complicated, but I think in theory it is possible that ``chunklen < chachelen``. Which will result in bug here

Reply:
It is safe to use since in the calling function it is checked.
However I also added check here, 'just in case'.